### PR TITLE
[screensaver.atv4] 1.5.0

### PR DESCRIPTION
**Pull request by enen92**
Moving the addon to krypton (it is in helix right now). I'll bump the other one with a minor revision to let users know further updates will be sent to krypton and above.

### CODE REVIEW

File: screensaver.atv4/entrychecksumgenerator.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+
+def tmpdownload(url):
+    file_name = url.split('/')[-1]
+    req = urllib2.request.Request(url)
+    u = urllib2.urlopen(req)
+
+    meta = u.info()
+    file_size = int(meta.getheaders("Content-Length")[0])
+    print "Downloading: %s Bytes: %s" % (file_name, file_size)
```
Comment (Rechi) [inline screensaver.atv4/entrychecksumgenerator.py]: `xbmc.log`

File: screensaver.atv4/entrychecksumgenerator.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    with open(os.path.join(tmpfolder, file_name), 'wb') as f:
+        while True:
+            _buffer = u.read(block_sz)
+            if not _buffer:
+                return file_name
+            file_size_dl += len(_buffer)
+            f.write(_buffer)
+            status = r"%10d [%3.2f%%]" % (file_size_dl, file_size_dl * 100. / file_size)
+            status = status + chr(8) * (len(status) + 1)
+            print status,
```
Comment (Rechi) [inline screensaver.atv4/entrychecksumgenerator.py]: `xbmc.log`

File: screensaver.atv4/entrychecksumgenerator.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        if not os.path.exists(tmpfolder):
+            os.mkdir(tmpfolder)
+        checksums = {}
+        failed = []
+
+        video_feed = json.loads(html)
+        for block in video_feed:
+            for asset in block["assets"]:
+                asset_url = asset["url"]
+                print "Processing video %s..." % (asset_url)
```
Comment (Rechi) [inline screensaver.atv4/entrychecksumgenerator.py]: `xbmc.log`

File: screensaver.atv4/entrychecksumgenerator.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                file_name = tmpdownload(asset_url)
+                if file_name:
+                    with open(os.path.join(tmpfolder, file_name), "rb") as f:
+                        checksum = hashlib.md5(f.read()).hexdigest()
+                        checksums[file_name] = checksum
+                        os.remove(os.path.join(tmpfolder, file_name))
+                        print "File processed. Checksum= %s" % (checksum)
```
Comment (Rechi) [inline screensaver.atv4/entrychecksumgenerator.py]: `xbmc.log`

File: screensaver.atv4/entrychecksumgenerator.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                else:
+                    failed.append(asset_url)
+
+        shutil.rmtree(tmpfolder)
+        print "Updating checksum file"
```
Comment (Rechi) [inline screensaver.atv4/entrychecksumgenerator.py]: `xbmc.log`

File: screensaver.atv4/entrychecksumgenerator.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        with open(os.path.join("resources", "checksums.json"), "w") as f:
+            f.write(json.dumps(checksums))
+            print "All done"
```
Comment (Rechi) [inline screensaver.atv4/entrychecksumgenerator.py]: `xbmc.log`

File: screensaver.atv4/entrychecksumgenerator.py [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        print "Failed items: %s" % (str(failed))
```
Comment (Rechi) [inline screensaver.atv4/entrychecksumgenerator.py]: `xbmc.log`

File: screensaver.atv4/entrychecksumgenerator.py [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+    if response.getcode() == 200:
+        locations = []
+        html = response.read()
+        video_feed = json.loads(html)
+        for block in video_feed:
+            for asset in block["assets"]:
+                if asset["accessibilityLabel"].lower() not in locations:
+                    locations.append(asset["accessibilityLabel"].lower())
+        print locations
```
Comment (Rechi) [inline screensaver.atv4/entrychecksumgenerator.py]: `xbmc.log`

**Comment by enen92** [comment]
@Rechi the file is not used in kodi, it is just an helper file to generate the checksums of videos for download

**Comment by Lunatixz** [comment]
LGTM

**Pull request merged**